src/psyclone/dynamo0p3_builtins.py

```diff
@@ -281,84 +360,157 @@ def gen_code(self, parent):
         from f2pygen import AssignGen
         # We subtract each element of f2 from the corresponding element
         # of f1 and store the result in f3.
-        invar_name1 = self.array_ref(self._arguments.args[0].proxy_name)
-        invar_name2 = self.array_ref(self._arguments.args[1].proxy_name)
-        outvar_name = self.array_ref(self._arguments.args[2].proxy_name)
-        assign = AssignGen(parent, lhs=outvar_name,
-                           rhs=invar_name1 + " - " + invar_name2)
+        field_name3 = self.array_ref(self._arguments.args[0].proxy_name)
+        field_name1 = self.array_ref(self._arguments.args[1].proxy_name)
+        field_name2 = self.array_ref(self._arguments.args[2].proxy_name)
+        assign = AssignGen(parent, lhs=field_name3,
+                           rhs=field_name1 + " - " + field_name2)
         parent.add(assign)
 
 
-class DynAddFieldsKern(DynBuiltIn):
-    ''' Add one field to another and return the result as a third field '''
+class DynIncXMinusYKern(DynBuiltIn):
+    ''' Subtract the second field from the first field and return it '''
 
     def __str__(self):
-        return "Built-in: Add fields"
+        return "Built-in: Decrement field"
 
     def gen_code(self, parent):
         from f2pygen import AssignGen
-        # We add each element of f2 to the corresponding element of f1
-        # and store the result in f3.
-        invar_name1 = self.array_ref(self._arguments.args[0].proxy_name)
-        invar_name2 = self.array_ref(self._arguments.args[1].proxy_name)
-        outvar_name = self.array_ref(self._arguments.args[2].proxy_name)
-        parent.add(AssignGen(parent, lhs=outvar_name,
-                             rhs=invar_name1 + " + " + invar_name2))
+        # We subtract each element of f1 from the corresponding element of f2
+        # and store the result back in f1.
+        field_name1 = self.array_ref(self._arguments.args[0].proxy_name)
+        field_name2 = self.array_ref(self._arguments.args[1].proxy_name)
+        parent.add(AssignGen(parent, lhs=field_name1,
+                             rhs=field_name1 + " - " + field_name2))
 
 
-class DynDivideFieldsKern(DynBuiltIn):
-    ''' Divide the first field by the second and return the result as
-    a third field '''
+class DynAXMinusYKern(DynBuiltIn):
+    ''' Z = a.X - Y where 'a' is a scalar and 'Z', 'X' and
+    'Y' are fields '''
 
     def __str__(self):
-        return "Built-in: Divide fields"
+        return "Built-in: aX_minus_Y"
 
     def gen_code(self, parent):
         from f2pygen import AssignGen
```

Comment:
This is not your fault (it's mine) but, while you're here, please could you modify all of the `from f2pygen` imports to say `from psyclone.f2pygen ...`.

Reply:
Modified :-)